lib/registration.pm

```diff
@@ -215,8 +215,11 @@ sub fill_in_registration_data {
             assert_screen($modules_needle);
             # Add desktop module for SLES if desktop doesn't match default
             if (check_var('SLE_PRODUCT', 'sles') && (my $addons = get_var('SCC_ADDONS')) !~ /(?:desktop|we|productivity)/) {
```

Comment:
Seems the previous code was incorrect: we shouldn't make 'deskop' module as default for sles product.
In SLE 15, the default modules for sles are:
- Basesystem
- Server-Applications

The desktop module should be added to sles only if we really want gnome desktop environment to be installed.
Just like I did for addon_products_sle.pm: 
    push @addons, 'desktop' if check_var('DESKTOP', 'gnome') && !grep(/^desktop$/, @addons);

If HA testing doesn't need desktop module, it could work with DESKTOP=textmode

Reply:
I know :) but this has been discuss with @okurz and as Desktop was added by default in previous beta versions of SLE15, it may break lot of test to remove it.
This is why the change has been done only for HA.

Reply:
With your change, the desktop will never be installed on SLES15, right?
But we have some migration test cases to upgrade SLES12SP3+HA to SLES15+HA, considered that gnome is default on SLE12SP3, we also hope it is still installed on SLES15.

Reply:
No, if you read the code carefully (and I didn't make a mistake) the old code added the desktop addon to SLE if it wasn't already set in SCC_ADDONS.
My change only disable this forced addition for HA, but if you have `SCC_ADDONS=desktop,ha` in your test variables, it will honor it and add the desktop module.

Reply:
In upgrade from SLES12-SP3+HA to SLES15+HA, I have to use "SCC_ADDONS=ha". The 'desktop' can NOT be involved because it is not valid addon of SLE12. (Before upgrade, the original system, SLE12-SP3, should be registered and patched, desktop is not valid module at the time).

Reply:
I don't understand, in your first comment you said that 'desktop' module shouldn't be added by default in SLE15 and now you say that you need it? ;-)
And how your upgrade tests are done? You say that before upgrade you have a a fully patched SLE12-SP3, but do you use an image for patching to SLE15 or do you do a SLE12-SP3 installation **and** a SLE15 upgrade in the same test? If you use upgrade from an image, `SCC_ADDONS=desktop,ha` should be a valid option.

Or do you have a specific variable for the upgrade test? I could test it and let the code performs like the old behavior and only remove 'desktop' module for my case.

I need to remove 'desktop' for HA build validation, this is asking by the PR/RM.

Reply:
It's not conflict: desktop should not be installed by default on sles15 unless user explicitly enables it, for example with variable DESKTOP=gnome

The upgrade test (i.e. SLES12-SP3+HA to SLES15+HA) looks like:
1, boot an existing sles12-sp3 hdd image (gnome installed as default)
2, register to scc and get system updated (also register addons listed in SCC_ADDONS)
    In this case, SCC_ADDONS=ha. 
    (de-register from scc if media based upgrade)
3, reboot with sle15 installer image
4, do upgrade (manually select sle15 extensions and modules if media based upgrade)

No specific variable for the upgrade test, it uses SCC_ADDONS to enable extensions or modules.

I understand that your change is critical for HA build validation, it's fine to me with your change. I can take care of the issue in upgrade testing later.

Reply:
Ok, I will have a look further on the code, but it should enable 'desktop' module if `DESKTOP=gnome`. My change should not break anything :)

Another question: it's a little bit strange that HA upgrade is testing with `DESKTOP=gnome`, a HA server is more installed in minimal or textmode, rarely in graphical mode. Is it mandatory for you? Asked by Project/Release Manager? (this could be discussed apart from this MR :-))

Reply:
We try to upgrade SLE with different addons, i.e. sles12-sp3+ha to sles15+ha, but not ha specific function test is actually involved in upgrading test.

Not mandatory. I only get info from PM that ha must be covered, without info about gnome or textmode.  I think gnome is included just because it is default to sle12.

Actually in upgrading test, the more packages installed on system, the easier to fail. So I think if sle12-gnome+ha succeeds to upgrade to sle15, it should also succeed on sle12-textmode+ha. But it is not true on the other hand.